testbed/tests/window/test_window.py

```diff
@@ -138,13 +139,194 @@ async def test_move_and_resize(main_window, main_window_probe, capsys):
         finally:
             main_window.content = orig_content
 
-    async def test_full_screen(main_window, main_window_probe):
-        """Window can be made full screen"""
-        main_window.full_screen = True
-        await main_window_probe.wait_for_window("Full screen is a no-op")
+    @pytest.mark.parametrize(
+        "initial_state, final_state",
+        [
+            # Direct switch from NORMAL:
+            (WindowState.NORMAL, WindowState.FULLSCREEN),
+            (WindowState.NORMAL, WindowState.PRESENTATION),
+            # Direct switch from FULLSCREEN:
+            (WindowState.FULLSCREEN, WindowState.NORMAL),
+            (WindowState.FULLSCREEN, WindowState.PRESENTATION),
+            # Direct switch from PRESENTATION:
+            (WindowState.PRESENTATION, WindowState.NORMAL),
+            (WindowState.PRESENTATION, WindowState.FULLSCREEN),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "intermediate_states",
+        [
+            (
+                WindowState.FULLSCREEN,
+                WindowState.PRESENTATION,
+                WindowState.NORMAL,
+                WindowState.FULLSCREEN,
+                WindowState.NORMAL,
+                WindowState.PRESENTATION,
+            ),
+            (
+                WindowState.PRESENTATION,
+                WindowState.FULLSCREEN,
+                WindowState.NORMAL,
+                WindowState.PRESENTATION,
+                WindowState.NORMAL,
+                WindowState.FULLSCREEN,
+            ),
+        ],
+    )
+    async def test_window_state_change_with_intermediate_states(
+        app,
+        initial_state,
+        final_state,
+        main_window,
+        main_window_probe,
+        intermediate_states,
+    ):
+        """Window state can be directly changed to another state."""
+        if not main_window_probe.supports_fullscreen and WindowState.FULLSCREEN in {
+            initial_state,
+            final_state,
+        }:
+            pytest.xfail("This backend doesn't support fullscreen window state.")
+        if (
+            not main_window_probe.supports_presentation
+            and WindowState.PRESENTATION
+            in {
+                initial_state,
+                final_state,
+            }
+        ):
+            pytest.xfail("This backend doesn't support presentation window state.")
+
+        # Set to initial state
+        main_window.state = initial_state
+        await main_window_probe.wait_for_window(f"Main window is in {initial_state}")
+
+        assert main_window.state == initial_state
```

Comment:
With the change to `get_window_state()`, this now requires a check of the backend with "actual=False" so that we get actual window data, not just the pending state.

Reply:
I have added a new property on the window probe `window_probe.instantaneous_state` that reports the actual window state and not the pending state. Although both instantaneous_state and state will be same on the Android backend.